Archive pruner

smruthi2187 opened the pull request:
Archive Pruner is  meant to clean up orphaned archives from storage.
By default, it's configured to run every hour. The value can be set in Values.yaml to any preferred interval.

README.md under storagesvc package gives an overview of the changes made to this package.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/fission/fission/455)
<!-- Reviewable:end -->

life1347 commented:
*[storagesvc/crdClient.go, line 15 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3YyfAqnUzH_h1LyopY:-L3YyfAqnUzH_h1LyopZ:b-gkm45p) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/storagesvc/crdClient.go#L15)):*
> ```Go
> 	fissionClient, _, _, err := crd.MakeFissionClient()
> 	if err != nil {
> 		return nil
> ```

We should handle the error instead of return nil here, or we may encounter the nil pointer problem.
How about adding `log.Fatalf` to print out the error message?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fission/fission/455)*
<!-- Sent from Reviewable.io -->

life1347 commented:
*[executor/poolmgr/gp.go, line 306 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3YrG9yznwilu_0LXL_:-L3YrG9yznwilu_0LXLa:bfw6xmt) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/executor/poolmgr/gp.go#L306)):*
> ```Go
> 	if isv6 == false {
> 		baseUrl = fmt.Sprintf("http://%v:8000/", podIP)
> 		log.Printf("ipv6 fasle : baseUrl : %s", baseUrl)
> ```

typo: `ipv6 false`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fission/fission/455)*
<!-- Sent from Reviewable.io -->

life1347 commented:
*[storagesvc/storagesvc.go, line 23 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3YzqcPyUpOUVBNDkRu:-L3YzqcPyUpOUVBNDkRv:b-8ukxtu) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/storagesvc/storagesvc.go#L23)):*
> ```Go
> 	"errors"
> 	"fmt"
> 	log "github.com/sirupsen/logrus"
> ```

just a minor issue, we try to keep the import in some order, please refer: https://github.com/fission/fission/pull/210#issuecomment-304515616

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fission/fission/455)*
<!-- Sent from Reviewable.io -->

life1347 commented:
*[storagesvc/util.go, line 8 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3Yw76NVsXZPMJM2IIl:-L3YvG5ON2DqRddZj8pf:bvm3f6t) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/storagesvc/util.go#L8)):*
> func utilGetQueryParamValue(urlString string, queryParam string) (string, error) {

It will be better to return both error and string and let the caller handle the error instead of simply return an empty string.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fission/fission/455)*
<!-- Sent from Reviewable.io -->

life1347 commented:
*[storagesvc/util.go, line 11 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3YxBfk4gNNTZJkenGY:-L3YxBfk4gNNTZJkenGZ:b-pg5g8d) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/storagesvc/util.go#L11)):*
> ```Go
> 	url, err := url.Parse(urlString)
> 	if err != nil {
> 		log.Printf("Error parsing URL string: %s into URL", urlString)
> ```

do early return here, let the caller handle the error.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fission/fission/455)*
<!-- Sent from Reviewable.io -->

life1347 commented:
*[storagesvc/stowClient.go, line 5 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3Z1Lh9MzjvbXtKMwBX:-L3Z1Lh9MzjvbXtKMwBY:bkr2tnf) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/storagesvc/stowClient.go#L5)):*
> ```Go
> import (
> 	"errors"
> 	"github.com/graymeta/stow"
> ```

It will be great if we can sort imports here.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fission/fission/455)*
<!-- Sent from Reviewable.io -->

life1347 commented:
*[charts/fission-all/templates/deployment.yaml, line 491 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3Z3taMHTkGnbiR6Klu:-L3Z3taMHTkGnbiR6Klv:bmk7j2j) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/charts/fission-all/templates/deployment.yaml#L491)):*
> ```YAML
>         env:
>         - name: PRUNE_INTERVAL
>           value: "{{.Values.pruneInterval}}"
> ```

Also need change in charts/fission-core/templates/deployment.yaml

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fission/fission/455)*
<!-- Sent from Reviewable.io -->

life1347 commented:
*[charts/fission-all/values.yaml, line 73 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3Z48Q8xeo8-C7vxM47:-L3Z48Q8xeo8-C7vxM48:b-m9p5jf) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/charts/fission-all/values.yaml#L73)):*
> ```YAML
> ## This interval configures the frequency at which it runs inside the storagesvc pod.
> ## The value is in minutes.
> pruneInterval: 60
> ```

Also need the change in charts/fission-core/values.yaml

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fission/fission/455)*
<!-- Sent from Reviewable.io -->

life1347 commented:
*[storagesvc/storagesvc.go, line 134 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3Z4qMyxIlX7H21TJAG:-L3Z4qMyxIlX7H21TJAH:bjtc7rl) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/storagesvc/storagesvc.go#L134)):*
> ```Go
> 
> 	// Get the file (called "item" in stow's jargon), open it,
> 	// stream it to response
> ```

The comment should also move to StowClient.getFileIntoResponse()

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fission/fission/455)*
<!-- Sent from Reviewable.io -->

life1347 commented:
Reviewed 1 of 13 files at r1.
Review status: 1 of 13 files reviewed at latest revision, 9 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fission/fission/455)*
<!-- Sent from Reviewable.io -->

life1347 commented:
Hi @smruthi2187, still reviewing, and there are some comments so far.
And, about the flag to list orphan pkgs like `fission package list --orphan` we discussed on the slack will be added to this PR or a new one?

smruthi2187 commented:
Thanks for reviewing Ta-Ching. I'll address your comments soon. About the new CLI, I'll run it by Soam and give you an update. However, it'll be part of a separate review request. 

---

Review status: 1 of 13 files reviewed at latest revision, 9 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fission/fission/455#-:-L3ZBZZz3Miqao2yGhsw:bacyki2)*
<!-- Sent from Reviewable.io -->

prithviramesh commented:
Review status: 1 of 13 files reviewed at latest revision, 16 unresolved discussions.

---

*[environments/fetcher/fetcher.go, line 133 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3ZJtqGguUf-WVht-S7:-L3ZJtqGguUf-WVht-S8:bf4snvx) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/environments/fetcher/fetcher.go#L133)):*
> ```Go
> 
> func (fetcher *Fetcher) FetchHandler(w http.ResponseWriter, r *http.Request) {
> 	log.Println("just got in fetch handler")
> ```

These log messages seem like they're more for (your) development/debugging, do we need these as part of the final PR?

---

*[environments/fetcher/fetcher.go, line 145 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3ZK6qHDnPSzcb7aVw5:-L3ZK6qHDnPSzcb7aVw6:bbxgkzp) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/environments/fetcher/fetcher.go#L145)):*
> ```Go
> 	}()
> 
> 	log.Println("parsing request")
> ```

Same comment as above

---

*[executor/poolmgr/gp.go, line 347 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3ZKFxpyZVlWHux82gh:-L3ZKFxpyZVlWHux82gi:bra25ad) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/executor/poolmgr/gp.go#L347)):*
> ```Go
> 	// tell fetcher to get the function.
> 	fetcherUrl := gp.getFetcherUrl(podIP)
> 	log.Printf("fetcherUrl : %s", fetcherUrl)
> ```

same comment as above about the log messages

---

*[storagesvc/archivePruner.go, line 75 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3ZLgYpI96hvKy8Mlkl:-L3ZLgYpI96hvKy8Mlkm:b-p0pbon) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/storagesvc/archivePruner.go#L75)):*
> ```Go
> 	}
> 
> 	log.WithField("list", "archives referenced by packages").Debugf("%s", archivesRefByPkgs)
> ```

Same comment about logging as above

---

*[storagesvc/archivePruner.go, line 86 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3ZM-fIFl6EL9DMzkym:-L3ZM-fIFl6EL9DMzkyn:b-ug4y6p) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/storagesvc/archivePruner.go#L86)):*
> ```Go
> 		return
> 	}
> 	log.WithField("list", "archives in storage").Debugf("%s", archivesInStorage)
> ```

same comment as above about logging 

---

*[storagesvc/archivePruner.go, line 91 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3ZMD1ZzTlbvNkO5VDB:-L3ZMD1ZzTlbvNkO5VDC:b-ug4y6p) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/storagesvc/archivePruner.go#L91)):*
> ```Go
> 	// need to do something more optimal at scale.
> 	orphanedArchives := utilGetDifferenceOfLists(archivesInStorage, archivesRefByPkgs)
> 	log.WithField("list", "orphan archives").Debugf("%s", orphanedArchives)
> ```

same comment as above about logging 

---

*[storagesvc/storagesvc.go, line 69 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3ZOdLB8sn1AFgsr8j-:-L3ZOdLB8sn1AFgsr8j0:b-kl8g4c) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/storagesvc/storagesvc.go#L69)):*
> ```Go
> 	fileSize, err := strconv.Atoi(fileSizeS[0])
> 	if err != nil {
> 		log.WithError(err).Errorf("Error parsing x-file-size: '%v'", fileSizeS)
> ```

good catch!

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fission/fission/455)*
<!-- Sent from Reviewable.io -->

smruthi2187 commented:
Review status: 1 of 13 files reviewed at latest revision, 16 unresolved discussions.

---

*[environments/fetcher/fetcher.go, line 133 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3ZJtqGguUf-WVht-S7:-L3Zm6aHnRx22yIUDXZC:b-rb13j5) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/environments/fetcher/fetcher.go#L133)):*
<details><summary><i>Previously, prithviramesh wrote…</i></summary><blockquote>

These log messages seem like they're more for (your) development/debugging, do we need these as part of the final PR?
</blockquote></details>

Yes, this file , fetcher.go and the other file, gp.go shouldn't have been part of this PR at all. I was tracing some flow, so added these prints there. My bad I `git commit`ed them. I'll remove them from this PR. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fission/fission/455)*
<!-- Sent from Reviewable.io -->

soamvasani commented:
Reviewed 7 of 13 files at r1.
Review status: 7 of 13 files reviewed at latest revision, 35 unresolved discussions.

---

*[charts/fission-all/values.yaml, line 70 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3ZCnm5yQFmf4ix34nd:-L3ZCnm6KuZfJYQzrFHp:bkw6r3i) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/charts/fission-all/values.yaml#L70)):*
> ```YAML
> analytics: true
> 
> ## archive pruner is a garbage collector for archives on the storage.
> ```

These comments are user-visible; please capitalize the first word.

Also, suggest "...archives on the fission storage service" or something like that.

---

*[environments/fetcher/fetcher.go, line 133 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3ZD61ti-8k4t0eTf7Q:-L3ZD61ti-8k4t0eTf7R:bgrq9c1) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/environments/fetcher/fetcher.go#L133)):*
> ```Go
> 
> func (fetcher *Fetcher) FetchHandler(w http.ResponseWriter, r *http.Request) {
> 	log.Println("just got in fetch handler")
> ```

remove, doesn't seem useful

---

*[environments/fetcher/fetcher.go, line 145 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3ZD8I3PMV7HZgA48b1:-L3ZD8I3PMV7HZgA48b2:bgrq9c1) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/environments/fetcher/fetcher.go#L145)):*
> ```Go
> 	}()
> 
> 	log.Println("parsing request")
> ```

remove, doesn't seem useful

---

*[executor/poolmgr/gp.go, line 347 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3ZDDENh7HITtKHKnvE:-L3ZDDEOHFlQwg_wSRFL:b-l6mwxj) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/executor/poolmgr/gp.go#L347)):*
> ```Go
> 	// tell fetcher to get the function.
> 	fetcherUrl := gp.getFetcherUrl(podIP)
> 	log.Printf("fetcherUrl : %s", fetcherUrl)
> ```

remove

---

*[storagesvc/archivePruner.go, line 1 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3ZDIB3qvHb9rFJkgYP:-L3ZDIB3qvHb9rFJkgYQ:bwbybcm) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/storagesvc/archivePruner.go#L1)):*
> ```Go
> package storagesvc
> ```

please add copyright header

---

*[storagesvc/archivePruner.go, line 5 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3ZDHF2hWMd1mTQhXn4:-L3ZDHF2hWMd1mTQhXn5:b-4r8ma4) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/storagesvc/archivePruner.go#L5)):*
> ```Go
> import (
> 	log "github.com/sirupsen/logrus"
> 	"time"
> ```

We have a convention of organizing our imports into 3 sections: first the standard library imports, then 3rd party non-stdlib stuff, then fission imports.

---

*[storagesvc/archivePruner.go, line 12 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3ZDZ9NIFhizFoyX3dy:-L3ZDZ9OYcFUPt4j6-v6:b-si7tl4) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/storagesvc/archivePruner.go#L12)):*
> ```Go
> 	archiveChan   chan (string)
> 	stowClient    *StowClient
> 	pruneInterval int
> ```

use a time.Duration

---

*[storagesvc/archivePruner.go, line 28 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3ZDnSvRchu1n2N7KAB:-L3ZDnSvRchu1n2N7KAC:biq4erx) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/storagesvc/archivePruner.go#L28)):*
> ```Go
> // This method listens to archiveChannel for archive ids that need to be deleted
> func (pruner *ArchivePruner) pruneArchives() {
> 	log.Info("listening to archiveChannel to prune archives..")
> ```

typo, double ".." at the end

---

*[storagesvc/archivePruner.go, line 36 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3ZDw3eoYHH4VVQyC3S:-L3ZDw3eoYHH4VVQyC3T:b-wy0ah5) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/storagesvc/archivePruner.go#L36)):*
> ```Go
> 				// logging the error and continuing with other deletions.
> 				// hopefully this archive will be deleted in the next iteration.
> 				log.WithField("archive ID", archiveID).WithError(err).Error("Error deleting archive")
> ```

can you say "ignoring error while deleting archive"

---

*[storagesvc/archivePruner.go, line 60 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3ZEoRzM0cyD3JSlnKj:-L3ZEoRzM0cyD3JSlnKk:b9mtsst) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/storagesvc/archivePruner.go#L60)):*
> ```Go
> 		// Safe to just silence the error here. Hoping next iteration will succeed.
> 		log.WithError(err).Error("Error getting package list from kubernetes")
> 		return
> ```

might be better to return the error and let the caller decide whether to ignore it?

---

*[storagesvc/crdClient.go, line 8 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3_BoVgFjLtjNF10z-t:-L3_BoVgFjLtjNF10z-u:b-42fxmb) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/storagesvc/crdClient.go#L8)):*
> ```Go
> )
> 
> type CRDClient struct {
> ```

I don't really see the value of this wrapper.  All it does is add a method for accessing .Items in a package list.

---

*[storagesvc/README.md, line 1 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3_C86rVpIrAh3tNPnZ:-L3_C86rVpIrAh3tNPn_:buxhfsv) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/storagesvc/README.md#L1)):*
> ```Markdown
> # StorageSvc 
> ```

:+1: This is great, we should have READMEs for more of the major components, it will make life easier for future devs.

---

*[storagesvc/storagesvc.go, line 83 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3_ChOtaV-qy5Z8tMNM:-L3_ChOu-1_x4NT2ZTvG:b-ahbobh) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/storagesvc/storagesvc.go#L83)):*
> ```Go
> 	if err != nil {
> 		log.WithError(err).Error("Error saving uploaded file")
> 		http.Error(w, "Error saving uploaded file", 500) // TODO : This was 400. I think it shd be 500.  TBD
> ```

Yes indeed, 500 is a better choice. (You can remove the TODO.)

---

*[storagesvc/stowClient.go, line 40 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3_DnP5PpF6JDN_66mk:-L3_DnP63AY_tP9fcG3h:bqrs5pl) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/storagesvc/stowClient.go#L40)):*
> ```Go
> var (
> 	ErrNotFound                = errors.New("not found")
> 	ErrRetrievingItem          = errors.New("not able to retrieve item")
> ```

unable* to

---

*[storagesvc/stowClient.go, line 94 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3_EBt1ZpMpo8PZHFQE:-L3_EBt1ZpMpo8PZHFQF:b-d0aeuc) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/storagesvc/stowClient.go#L94)):*
> ```Go
> }
> 
> // This method writes the file on the storage
> ```

godoc convention is to start the comment with the name of the function, so: "putFile writes..."

---

*[storagesvc/stowClient.go, line 112 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3_ETRdLwXxh6WTnRlR:-L3_ETRdLwXxh6WTnRlS:bout9ni) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/storagesvc/stowClient.go#L112)):*
> ```Go
> 
> // This method gets the file contents into http response
> func (client *StowClient) getFileIntoResponse(fileId string, w http.ResponseWriter) error {
> ```

I think this function is a bit odd. It copies things into an http.ResponseWriter but doesn't call http.Error. I mean, it's halfway HTTP-aware. How about accepting an io.Writer instead of an http.ResponseWriter and calling the function something like `copyFileToStream`

---

*[storagesvc/util.go, line 8 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3_FnGnNTZPC0qNKMeG:-L3_FnGnNTZPC0qNKMeH:b-g9pvie) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/storagesvc/util.go#L8)):*
> ```Go
> )
> 
> func utilGetQueryParamValue(urlString string, queryParam string) string {
> ```

please remove the "util" prefix

---

*[storagesvc/util.go, line 16 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3_GdGSK5d3AIA-tr02:-L3_GdGSK5d3AIA-tr03:b-bb0i16) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/storagesvc/util.go#L16)):*
> ```Go
> }
> 
> func utilGetDifferenceOfLists(firstList []string, secondList []string) []string {
> ```

please remove the util prefix

---

*[storagesvc/util.go, line 38 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3_FYmFoR4rf5xpgu-F:-L3_FYmGdvpaxAsybzcH:bxq44ds) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/storagesvc/util.go#L38)):*
> ```Go
> }
> 
> func utilDumpListContents(list []string, listDescription string) {
> ```

Remove this, doesn't look useful. Caller could just use log.Printf.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fission/fission/455)*
<!-- Sent from Reviewable.io -->

smruthi2187 commented:
Review status: 2 of 12 files reviewed at latest revision, 35 unresolved discussions, some commit checks failed.

---

*[charts/fission-all/values.yaml, line 70 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3ZCnm5yQFmf4ix34nd:-L3aY18g0P0SJw6CYqoW:b-896fix) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/charts/fission-all/values.yaml#L70)):*
<details><summary><i>Previously, soamvasani (Soam Vasani) wrote…</i></summary><blockquote>

These comments are user-visible; please capitalize the first word.

Also, suggest "...archives on the fission storage service" or something like that.
</blockquote></details>

Done.

---

*[charts/fission-all/values.yaml, line 73 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3Z48Q8xeo8-C7vxM47:-L3aY-Pu7NVGysufoqnb:b-896fix) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/charts/fission-all/values.yaml#L73)):*
<details><summary><i>Previously, life1347 (Ta-Ching Chen) wrote…</i></summary><blockquote>

Also need the change in charts/fission-core/values.yaml
</blockquote></details>

Done.

---

*[charts/fission-all/templates/deployment.yaml, line 491 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3Z3taMHTkGnbiR6Klu:-L3aY2QCD9P-vZJQu2H-:b-896fix) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/charts/fission-all/templates/deployment.yaml#L491)):*
<details><summary><i>Previously, life1347 (Ta-Ching Chen) wrote…</i></summary><blockquote>

Also need change in charts/fission-core/templates/deployment.yaml
</blockquote></details>

Done.

---

*[storagesvc/archivePruner.go, line 1 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3ZDIB3qvHb9rFJkgYP:-L3aYBBwuYmu9x2MvPej:b-896fix) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/storagesvc/archivePruner.go#L1)):*
<details><summary><i>Previously, soamvasani (Soam Vasani) wrote…</i></summary><blockquote>

please add copyright header
</blockquote></details>

Done.

---

*[storagesvc/archivePruner.go, line 12 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3ZDZ9NIFhizFoyX3dy:-L3aYG0iQEiqJQdMEGmK:b-896fix) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/storagesvc/archivePruner.go#L12)):*
<details><summary><i>Previously, soamvasani (Soam Vasani) wrote…</i></summary><blockquote>

use a time.Duration
</blockquote></details>

Done.

---

*[storagesvc/archivePruner.go, line 28 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3ZDnSvRchu1n2N7KAB:-L3aYHdp3ljcg4nJ31v3:b-896fix) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/storagesvc/archivePruner.go#L28)):*
<details><summary><i>Previously, soamvasani (Soam Vasani) wrote…</i></summary><blockquote>

typo, double ".." at the end
</blockquote></details>

Done.

---

*[storagesvc/archivePruner.go, line 36 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3ZDw3eoYHH4VVQyC3S:-L3aYItMiS6ORjOLjHWV:b-896fix) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/storagesvc/archivePruner.go#L36)):*
<details><summary><i>Previously, soamvasani (Soam Vasani) wrote…</i></summary><blockquote>

can you say "ignoring error while deleting archive"
</blockquote></details>

Done.

---

*[storagesvc/archivePruner.go, line 60 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3ZEoRzM0cyD3JSlnKj:-L3aYKithxCrjgcE6Mik:b-896fix) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/storagesvc/archivePruner.go#L60)):*
<details><summary><i>Previously, soamvasani (Soam Vasani) wrote…</i></summary><blockquote>

might be better to return the error and let the caller decide whether to ignore it?
</blockquote></details>

Done.

---

*[storagesvc/archivePruner.go, line 75 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3ZLgYpI96hvKy8Mlkl:-L3aYN42DCnIMo7qQkwx:b-qxcdmn) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/storagesvc/archivePruner.go#L75)):*
<details><summary><i>Previously, prithviramesh wrote…</i></summary><blockquote>

Same comment about logging as above
</blockquote></details>

So, I put this in debug, while, the default log level is set to Info. In the future, if I want to debug something locally, I can just set the default log level to debug to see them.

---

*[storagesvc/archivePruner.go, line 86 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3ZM-fIFl6EL9DMzkym:-L3aYbOr0nwWr3FmHe1u:b-hy5qgf) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/storagesvc/archivePruner.go#L86)):*
<details><summary><i>Previously, prithviramesh wrote…</i></summary><blockquote>

same comment as above about logging 
</blockquote></details>

replied above

---

*[storagesvc/archivePruner.go, line 91 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3ZMD1ZzTlbvNkO5VDB:-L3aYcaxgHnZCbLUDiHr:b-hy5qgf) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/storagesvc/archivePruner.go#L91)):*
<details><summary><i>Previously, prithviramesh wrote…</i></summary><blockquote>

same comment as above about logging 
</blockquote></details>

replied above

---

*[storagesvc/storagesvc.go, line 23 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3YzqcPyUpOUVBNDkRu:-L3aYgDgej0iZpD2wGEX:b-896fix) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/storagesvc/storagesvc.go#L23)):*
<details><summary><i>Previously, life1347 (Ta-Ching Chen) wrote…</i></summary><blockquote>

just a minor issue, we try to keep the import in some order, please refer: https://github.com/fission/fission/pull/210#issuecomment-304515616
</blockquote></details>

Done.

---

*[storagesvc/storagesvc.go, line 69 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3ZOdLB8sn1AFgsr8j-:-L3aYhZBFrcEJX_W3YGQ:b3k473f) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/storagesvc/storagesvc.go#L69)):*
<details><summary><i>Previously, prithviramesh wrote…</i></summary><blockquote>

good catch!
</blockquote></details>

It was present before. I just moved the code around :)

---

*[storagesvc/storagesvc.go, line 83 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3_ChOtaV-qy5Z8tMNM:-L3aYnUgmqv1FCmT0AeN:b-896fix) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/storagesvc/storagesvc.go#L83)):*
<details><summary><i>Previously, soamvasani (Soam Vasani) wrote…</i></summary><blockquote>

Yes indeed, 500 is a better choice. (You can remove the TODO.)
</blockquote></details>

Done.

---

*[storagesvc/storagesvc.go, line 134 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3Z4qMyxIlX7H21TJAG:-L3aYpCo1jdUzAKkEWPS:b-plahc3) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/storagesvc/storagesvc.go#L134)):*
<details><summary><i>Previously, life1347 (Ta-Ching Chen) wrote…</i></summary><blockquote>

The comment should also move to StowClient.getFileIntoResponse()
</blockquote></details>

Changed this a bit in the new version.

---

*[storagesvc/stowClient.go, line 5 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3Z1Lh9MzjvbXtKMwBX:-L3aYsnjT2be5aAp0QU2:b-896fix) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/storagesvc/stowClient.go#L5)):*
<details><summary><i>Previously, life1347 (Ta-Ching Chen) wrote…</i></summary><blockquote>

It will be great if we can sort imports here.
</blockquote></details>

Done.

---

*[storagesvc/stowClient.go, line 40 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3_DnP5PpF6JDN_66mk:-L3aYurK0AQrVU-NPLJs:b-896fix) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/storagesvc/stowClient.go#L40)):*
<details><summary><i>Previously, soamvasani (Soam Vasani) wrote…</i></summary><blockquote>

unable* to
</blockquote></details>

Done.

---

*[storagesvc/stowClient.go, line 94 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3_EBt1ZpMpo8PZHFQE:-L3aYwMTDERByh0mm0ry:b-896fix) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/storagesvc/stowClient.go#L94)):*
<details><summary><i>Previously, soamvasani (Soam Vasani) wrote…</i></summary><blockquote>

godoc convention is to start the comment with the name of the function, so: "putFile writes..."
</blockquote></details>

Done.

---

*[storagesvc/stowClient.go, line 112 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3_ETRdLwXxh6WTnRlR:-L3aYwwohmanQ-zXLja3:b-se0ug5) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/storagesvc/stowClient.go#L112)):*
<details><summary><i>Previously, soamvasani (Soam Vasani) wrote…</i></summary><blockquote>

I think this function is a bit odd. It copies things into an http.ResponseWriter but doesn't call http.Error. I mean, it's halfway HTTP-aware. How about accepting an io.Writer instead of an http.ResponseWriter and calling the function something like `copyFileToStream`
</blockquote></details>

yeah, I agree. 

---

*[storagesvc/util.go, line 8 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3Yw76NVsXZPMJM2IIl:-L3aZ2X17RSE8jbiuhyY:b-896fix) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/storagesvc/util.go#L8)):*
<details><summary><i>Previously, life1347 (Ta-Ching Chen) wrote…</i></summary><blockquote>

> func utilGetQueryParamValue(urlString string, queryParam string) (string, error) {

It will be better to return both error and string and let the caller handle the error instead of simply return an empty string.

</blockquote></details>

Done.

---

*[storagesvc/util.go, line 8 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3_FnGnNTZPC0qNKMeG:-L3aZ3N4_pm_EdsFKtkL:b-896fix) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/storagesvc/util.go#L8)):*
<details><summary><i>Previously, soamvasani (Soam Vasani) wrote…</i></summary><blockquote>

please remove the "util" prefix
</blockquote></details>

Done.

---

*[storagesvc/util.go, line 11 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3YxBfk4gNNTZJkenGY:-L3aZ5GO3Trp-nWbxGbd:b-896fix) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/storagesvc/util.go#L11)):*
<details><summary><i>Previously, life1347 (Ta-Ching Chen) wrote…</i></summary><blockquote>

do early return here, let the caller handle the error.
</blockquote></details>

Done.

---

*[storagesvc/util.go, line 16 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3_GdGSK5d3AIA-tr02:-L3aZ684no5_9nP6xhxf:b-896fix) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/storagesvc/util.go#L16)):*
<details><summary><i>Previously, soamvasani (Soam Vasani) wrote…</i></summary><blockquote>

please remove the util prefix
</blockquote></details>

Done.

---

*[storagesvc/util.go, line 38 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3_FYmFoR4rf5xpgu-F:-L3aZ708KaCfASsIjZHU:b-896fix) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/storagesvc/util.go#L38)):*
<details><summary><i>Previously, soamvasani (Soam Vasani) wrote…</i></summary><blockquote>

Remove this, doesn't look useful. Caller could just use log.Printf.
</blockquote></details>

Done.

---

*[environments/fetcher/fetcher.go, line 133 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3ZD61ti-8k4t0eTf7Q:-L3aZ9884glWJL8tcOOy:b-4isyok) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/environments/fetcher/fetcher.go#L133)):*
<details><summary><i>Previously, soamvasani (Soam Vasani) wrote…</i></summary><blockquote>

remove, doesn't seem useful
</blockquote></details>

sorry these files fetcher.go and gp.go were not meant to be commited at all as part of this review. It was for my debugging. I've reverted them back to original version. 

---

*[environments/fetcher/fetcher.go, line 145 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3ZD8I3PMV7HZgA48b1:-L3aZQ2Xapa2CWKxPZ7B:b-896fix) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/environments/fetcher/fetcher.go#L145)):*
<details><summary><i>Previously, soamvasani (Soam Vasani) wrote…</i></summary><blockquote>

remove, doesn't seem useful
</blockquote></details>

Done.

---

*[environments/fetcher/fetcher.go, line 145 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3ZK6qHDnPSzcb7aVw5:-L3aZQZ04L049Ploz39p:b-896fix) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/environments/fetcher/fetcher.go#L145)):*
<details><summary><i>Previously, prithviramesh wrote…</i></summary><blockquote>

Same comment as above
</blockquote></details>

Done.

---

*[executor/poolmgr/gp.go, line 306 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3YrG9yznwilu_0LXL_:-L3aZRzxExibl35zkyAh:b-896fix) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/executor/poolmgr/gp.go#L306)):*
<details><summary><i>Previously, life1347 (Ta-Ching Chen) wrote…</i></summary><blockquote>

typo: `ipv6 false`
</blockquote></details>

Done.

---

*[executor/poolmgr/gp.go, line 347 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3ZDDENh7HITtKHKnvE:-L3aZSmLsrlsbWgQCv_Q:b-896fix) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/executor/poolmgr/gp.go#L347)):*
<details><summary><i>Previously, soamvasani (Soam Vasani) wrote…</i></summary><blockquote>

remove
</blockquote></details>

Done.

---

*[executor/poolmgr/gp.go, line 347 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3ZKFxpyZVlWHux82gh:-L3aZTVoKx4j2ogv5cSk:b-896fix) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/executor/poolmgr/gp.go#L347)):*
<details><summary><i>Previously, prithviramesh wrote…</i></summary><blockquote>

same comment as above about the log messages
</blockquote></details>

Done.

---

*[storagesvc/crdClient.go, line 8 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3_BoVgFjLtjNF10z-t:-L3aZYGmaEoCW6p46PiC:b-896fix) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/storagesvc/crdClient.go#L8)):*
<details><summary><i>Previously, soamvasani (Soam Vasani) wrote…</i></summary><blockquote>

I don't really see the value of this wrapper.  All it does is add a method for accessing .Items in a package list.
</blockquote></details>

Done.

---

*[storagesvc/crdClient.go, line 15 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3YyfAqnUzH_h1LyopY:-L3aZWAPFN9CcUuByFpn:bn4acke) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/storagesvc/crdClient.go#L15)):*
<details><summary><i>Previously, life1347 (Ta-Ching Chen) wrote…</i></summary><blockquote>

We should handle the error instead of return nil here, or we may encounter the nil pointer problem.
How about adding `log.Fatalf` to print out the error message?
</blockquote></details>

I've removed this file.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fission/fission/455)*
<!-- Sent from Reviewable.io -->

smruthi2187 commented:
Review status: 2 of 12 files reviewed at latest revision, 35 unresolved discussions, some commit checks failed.

---

*[storagesvc/archivePruner.go, line 5 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3ZDHF2hWMd1mTQhXn4:-L3aZazeOCcqsiuh5bos:b-896fix) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/storagesvc/archivePruner.go#L5)):*
<details><summary><i>Previously, soamvasani (Soam Vasani) wrote…</i></summary><blockquote>

We have a convention of organizing our imports into 3 sections: first the standard library imports, then 3rd party non-stdlib stuff, then fission imports.
</blockquote></details>

Done.

---

*[storagesvc/README.md, line 1 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3_C86rVpIrAh3tNPnZ:-L3aZdQiD0f69Vn6s767:b-zcjvte) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/storagesvc/README.md#L1)):*
<details><summary><i>Previously, soamvasani (Soam Vasani) wrote…</i></summary><blockquote>

:+1: This is great, we should have READMEs for more of the major components, it will make life easier for future devs.
</blockquote></details>

Thanks

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fission/fission/455)*
<!-- Sent from Reviewable.io -->

life1347 commented:
*[storagesvc/README.md, line 1 at r1](https://reviewable.io:443/reviews/fission/fission/455#-L3_C86rVpIrAh3tNPnZ:-L3aitRW1TmR-jAHi2qD:b4c9lyr) ([raw file](https://github.com/fission/fission/blob/2929d4fd5503fefffe25d42f855c1bd9ffd6c4e6/storagesvc/README.md#L1)):*
<details><summary><i>Previously, smruthi2187 wrote…</i></summary><blockquote>

Thanks
</blockquote></details>

:thumbsup: 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fission/fission/455)*
<!-- Sent from Reviewable.io -->

life1347 commented:
Reviewed 5 of 13 files at r1, 9 of 10 files at r2.
Review status: 11 of 12 files reviewed at latest revision, 35 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fission/fission/455)*
<!-- Sent from Reviewable.io -->

life1347 commented:
Reviewed 1 of 10 files at r2.
Review status: all files reviewed at latest revision, 35 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fission/fission/455)*
<!-- Sent from Reviewable.io -->

soamvasani commented:
LGTM

---

Reviewed 5 of 6 files at r3.
Review status: all files reviewed at latest revision, 24 unresolved discussions.

---

*[storagesvc/storagesvc.go, line 183 at r3](https://reviewable.io:443/reviews/fission/fission/455#-L3oa2Qi5Ki-ayjBN8pb:-L3oa2Qi5Ki-ayjBN8pc:bnbq2zl) ([raw file](https://github.com/fission/fission/blob/e49571e313426dc38289fa2e7ab88618621f9de1/storagesvc/storagesvc.go#L183)):*
> ```Go
> 	go storageService.Start(port)
> 
> 	// had to introduce this param to prevent storagesvc unit test from needing to talk to kubernetes
> ```

just a nit, "enablePruner prevents..."

---

*[test/build_and_test.sh, line 19 at r3](https://reviewable.io:443/reviews/fission/fission/455#-L3oalVAM0qOFF87u4Zg:-L3oalVBqn4kfW-ZUVmy:b-ykz7wq) ([raw file](https://github.com/fission/fission/blob/e49571e313426dc38289fa2e7ab88618621f9de1/test/build_and_test.sh#L19)):*
> ```Shell
> BUILDER_IMAGE=$REPO/builder
> TAG=test
> PRUNE_INTERVAL=1
> ```

please add a comment with units (minutes? seconds?)

---

*[test/tests/test_archive_pruner.sh, line 38 at r3](https://reviewable.io:443/reviews/fission/fission/455#-L3ob8tTTaFN2JvZ6SMY:-L3ob8tUebELzJjVWlyF:bo5r2jo) ([raw file](https://github.com/fission/fission/blob/e49571e313426dc38289fa2e7ab88618621f9de1/test/tests/test_archive_pruner.sh#L38)):*
> ```Shell
>     else
>         # Linux
>         truncate -s 512M test_dir/dynamically_generated_file
> ```

Can you make this smaller? 1M will be good enough, right?  I'm asking because Travis CI may get too slow and timeout while uploading 512M.

---

*[test/tests/test_archive_pruner.sh, line 48 at r3](https://reviewable.io:443/reviews/fission/fission/455#-L3ocG6JBfBSPYse0w25:-L3ocG6JBfBSPYse0w26:b-sogude) ([raw file](https://github.com/fission/fission/blob/e49571e313426dc38289fa2e7ab88618621f9de1/test/tests/test_archive_pruner.sh#L48)):*
> ```Shell
>     echo "Creating package"
>     pkg=$(fission package create --deploy "test-deploy-pkg.zip" --env python| cut -f2 -d' '| tr -d \')
>     check_return_status $?
> ```

Do you need this? Won't `set -euo pipefail` cause the function to exit if the `fission package...` command fails?

---

*[test/tests/test_archive_pruner.sh, line 54 at r3](https://reviewable.io:443/reviews/fission/fission/455#-L3od6kb73MAjxNQzfZC:-L3od6kb73MAjxNQzfZD:bf7i46n) ([raw file](https://github.com/fission/fission/blob/e49571e313426dc38289fa2e7ab88618621f9de1/test/tests/test_archive_pruner.sh#L54)):*
> ```Shell
>     echo "Deleting package: $1"
>     fission package delete --name $1
>     check_return_status $?
> ```

this can also be removed

---

*[test/tests/test_archive_pruner.sh, line 60 at r3](https://reviewable.io:443/reviews/fission/fission/455#-L3odBSrLKdfvJyMfxP4:-L3odBSrLKdfvJyMfxP5:bgyj0t4) ([raw file](https://github.com/fission/fission/blob/e49571e313426dc38289fa2e7ab88618621f9de1/test/tests/test_archive_pruner.sh#L60)):*
> ```Shell
>     echo "Getting archive URL from package: $1"
>     url=`kubectl get package $1 -ojsonpath='{.spec.deployment.url}'`
>     check_return_status $?
> ```

also remove

---

*[test/tests/test_archive_pruner.sh, line 86 at r3](https://reviewable.io:443/reviews/fission/fission/455#-L3obUoa9xzwoaudw9Gy:-L3obUobta2ws7idLBxj:b-is297b) ([raw file](https://github.com/fission/fission/blob/e49571e313426dc38289fa2e7ab88618621f9de1/test/tests/test_archive_pruner.sh#L86)):*
> ```Shell
> #7. now verify that both get deleted.
> #8. cleanup
> main() {
> ```

Safer to add `trap cleanup EXIT` somewhere here, and remove the call at the bottom.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fission/fission/455#-:-L3oeBPRKpAUPMe6oQvX:bb6ywmx)*
<!-- Sent from Reviewable.io -->

soamvasani commented:
This is close to done, so adding it to 0.5.0.

soamvasani commented:
Reviewed 1 of 6 files at r3.
Review status: all files reviewed at latest revision, 24 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fission/fission/455)*
<!-- Sent from Reviewable.io -->

smruthi2187 commented:
Review status: 6 of 14 files reviewed at latest revision, 24 unresolved discussions.

---

*[storagesvc/storagesvc.go, line 183 at r3](https://reviewable.io:443/reviews/fission/fission/455#-L3oa2Qi5Ki-ayjBN8pb:-L43iEp0bDCCwSnNvady:b-896fix) ([raw file](https://github.com/fission/fission/blob/e49571e313426dc38289fa2e7ab88618621f9de1/storagesvc/storagesvc.go#L183)):*
<details><summary><i>Previously, soamvasani (Soam Vasani) wrote…</i></summary><blockquote>

just a nit, "enablePruner prevents..."
</blockquote></details>

Done.

---

*[test/build_and_test.sh, line 19 at r3](https://reviewable.io:443/reviews/fission/fission/455#-L3oalVAM0qOFF87u4Zg:-L43iIoWnfWLymwVfNjh:b-896fix) ([raw file](https://github.com/fission/fission/blob/e49571e313426dc38289fa2e7ab88618621f9de1/test/build_and_test.sh#L19)):*
<details><summary><i>Previously, soamvasani (Soam Vasani) wrote…</i></summary><blockquote>

please add a comment with units (minutes? seconds?)
</blockquote></details>

Done.

---

*[test/tests/test_archive_pruner.sh, line 38 at r3](https://reviewable.io:443/reviews/fission/fission/455#-L3ob8tTTaFN2JvZ6SMY:-L43iLdgtG0oT7mfdeUX:b-896fix) ([raw file](https://github.com/fission/fission/blob/e49571e313426dc38289fa2e7ab88618621f9de1/test/tests/test_archive_pruner.sh#L38)):*
<details><summary><i>Previously, soamvasani (Soam Vasani) wrote…</i></summary><blockquote>

Can you make this smaller? 1M will be good enough, right?  I'm asking because Travis CI may get too slow and timeout while uploading 512M.
</blockquote></details>

Done.

---

*[test/tests/test_archive_pruner.sh, line 54 at r3](https://reviewable.io:443/reviews/fission/fission/455#-L3od6kb73MAjxNQzfZC:-L43iSB6rAdnc5IH8O5u:b-896fix) ([raw file](https://github.com/fission/fission/blob/e49571e313426dc38289fa2e7ab88618621f9de1/test/tests/test_archive_pruner.sh#L54)):*
<details><summary><i>Previously, soamvasani (Soam Vasani) wrote…</i></summary><blockquote>

this can also be removed
</blockquote></details>

Done.

---

*[test/tests/test_archive_pruner.sh, line 60 at r3](https://reviewable.io:443/reviews/fission/fission/455#-L3odBSrLKdfvJyMfxP4:-L43iSnA3AN8ktwGCTWP:b-896fix) ([raw file](https://github.com/fission/fission/blob/e49571e313426dc38289fa2e7ab88618621f9de1/test/tests/test_archive_pruner.sh#L60)):*
<details><summary><i>Previously, soamvasani (Soam Vasani) wrote…</i></summary><blockquote>

also remove
</blockquote></details>

Done.

---

*[test/tests/test_archive_pruner.sh, line 86 at r3](https://reviewable.io:443/reviews/fission/fission/455#-L3obUoa9xzwoaudw9Gy:-L43iVV_9AeImjyBqeAH:b-896fix) ([raw file](https://github.com/fission/fission/blob/e49571e313426dc38289fa2e7ab88618621f9de1/test/tests/test_archive_pruner.sh#L86)):*
<details><summary><i>Previously, soamvasani (Soam Vasani) wrote…</i></summary><blockquote>

Safer to add `trap cleanup EXIT` somewhere here, and remove the call at the bottom.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fission/fission/455)*
<!-- Sent from Reviewable.io -->

life1347 commented:
merged #471
